src/machine.rs

```diff
@@ -144,8 +148,14 @@ impl MemoryExtra {
         } else {
             None
         };
+        let data_race = if config.data_race_detector {
+            Some(Rc::new(data_race::GlobalState::new()))
+        }else{
```

Comment:
Please Ctrl-F for `}else` and make sure to add spaces (this still occurs a few times).